scripts/azureml-assets/azureml/assets/release/group_test.py

```diff
@@ -3,28 +3,57 @@
 """Python script to run group tests."""
 import argparse
 from subprocess import check_call
+from subprocess import run
 from pathlib import Path
 import azure.ai.ml
 from azure.ai.ml import MLClient
 from azure.identity import DefaultAzureCredential
+from collections import defaultdict
+import concurrent.futures
 import yaml
 import os
 import sys
+from azureml.assets.util import logger
 TEST_YML = "tests.yml"
 
+
+def run_pytest_job(job: Path, my_env: dict):
+    """Run single pytest job."""
+    p = run(f"pytest -q {job}", env=my_env, shell=True)
+    return_code = p.returncode
+    return return_code
+
+
+def run_pytest_jobs(pytest_jobs: dict, my_env: dict):
+    """Run multiple pytest jobs concurrently."""
+    print("Start running pytest jobs")
+    submitted_jobs = defaultdict(list)
+    executor = concurrent.futures.ThreadPoolExecutor()
```

Comment:
`ThreadPoolExecutor` should be used as part of a `with` statement. This function should be reworked to look more like this: https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor-example

`ThreadPoolExecutor` has a great way to receive completed futures (see `concurrent.futures.as_completed` from the page above) and should be used instead of polling as you are in the block starting on line 114.